boards/px4/fmu-v5/src/spi.cpp

```diff
@@ -35,29 +35,56 @@
 #include <drivers/drv_sensor.h>
 #include <nuttx/spi/spi.h>
 
-constexpr px4_spi_bus_t px4_spi_buses[SPI_BUS_MAX_BUS_ITEMS] = {
-	initSPIBus(SPI::Bus::SPI1, {
-		initSPIDevice(DRV_IMU_DEVTYPE_ICM20689, SPI::CS{GPIO::PortF, GPIO::Pin2}, SPI::DRDY{GPIO::PortB, GPIO::Pin4}),
-		initSPIDevice(DRV_IMU_DEVTYPE_ICM20602, SPI::CS{GPIO::PortF, GPIO::Pin3}, SPI::DRDY{GPIO::PortC, GPIO::Pin5}),
-		initSPIDevice(DRV_GYR_DEVTYPE_BMI055, SPI::CS{GPIO::PortF, GPIO::Pin4}, SPI::DRDY{GPIO::PortB, GPIO::Pin14}),
-		initSPIDevice(DRV_ACC_DEVTYPE_BMI055, SPI::CS{GPIO::PortG, GPIO::Pin10}, SPI::DRDY{GPIO::PortB, GPIO::Pin15}),
-	}, {GPIO::PortE, GPIO::Pin3}),
-	initSPIBus(SPI::Bus::SPI2, {
-		initSPIDevice(SPIDEV_FLASH(0), SPI::CS{GPIO::PortF, GPIO::Pin5})
+constexpr px4_spi_bus_all_hw_t px4_spi_buses_all_hw[BOARD_NUM_SPI_CFG_HW_VERSIONS] = {
+	initSPIHWVersion(HW_VER_REV(0, 0), {
+		initSPIBus(SPI::Bus::SPI1, {
+			initSPIDevice(DRV_IMU_DEVTYPE_ICM20689, SPI::CS{GPIO::PortF, GPIO::Pin2}, SPI::DRDY{GPIO::PortB, GPIO::Pin4}),
+			initSPIDevice(DRV_IMU_DEVTYPE_ICM20602, SPI::CS{GPIO::PortF, GPIO::Pin3}, SPI::DRDY{GPIO::PortC, GPIO::Pin5}),
+			initSPIDevice(DRV_GYR_DEVTYPE_BMI055, SPI::CS{GPIO::PortF, GPIO::Pin4}, SPI::DRDY{GPIO::PortB, GPIO::Pin14}),
+			initSPIDevice(DRV_ACC_DEVTYPE_BMI055, SPI::CS{GPIO::PortG, GPIO::Pin10}, SPI::DRDY{GPIO::PortB, GPIO::Pin15}),
+		}, {GPIO::PortE, GPIO::Pin3}),
+		initSPIBus(SPI::Bus::SPI2, {
+			initSPIDevice(SPIDEV_FLASH(0), SPI::CS{GPIO::PortF, GPIO::Pin5})
+		}),
+		initSPIBus(SPI::Bus::SPI4, {
+			initSPIDevice(DRV_BARO_DEVTYPE_MS5611, SPI::CS{GPIO::PortF, GPIO::Pin10}),
+		}),
+		initSPIBusExternal(SPI::Bus::SPI5, {
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin4}, SPI::DRDY{GPIO::PortD, GPIO::Pin15}),
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin10}),
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin11})
+		}),
+		initSPIBusExternal(SPI::Bus::SPI6, {
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin6}),
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin7}),
+			initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin8})
+		}),
 	}),
-	initSPIBus(SPI::Bus::SPI4, {
-		initSPIDevice(DRV_BARO_DEVTYPE_MS5611, SPI::CS{GPIO::PortF, GPIO::Pin10}),
-	}),
-	initSPIBusExternal(SPI::Bus::SPI5, {
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin4}, SPI::DRDY{GPIO::PortD, GPIO::Pin15}),
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin10}),
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin11})
-	}),
-	initSPIBusExternal(SPI::Bus::SPI6, {
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin6}),
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin7}),
-		initSPIConfigExternal(SPI::CS{GPIO::PortI, GPIO::Pin8})
+
+	initSPIHWVersion(HW_VER_REV(0, 2), {
```

Comment:
```suggestion
	initSPIHWVersion(HW_VER_REV(5, 2), {
```

A duplicate is needed for the Nano with InitSPIHWVersion(HW_VER_REV(6, 2), {

Reply:
While Appling power to the unit, please capture the PE3, VDD_3V3_SENSORS, and PC5, PB4 on a scope and post the images here.

Reply:
The channels are PE3, VDD_3V3_SENSORS, PC5, PB4 in sequence. Details are here:[DSLogic-la-220315-100851.csv](https://github.com/PX4/PX4-Autopilot/files/8249777/DSLogic-la-220315-100851.csv)

![批注 2022-03-15 102617](https://user-images.githubusercontent.com/58416716/158296725-dd588b3a-285d-44ee-ba17-936bccadc7aa.jpg)
![批注 2022-03-15 102506](https://user-images.githubusercontent.com/58416716/158296654-125b2c95-3062-4d53-aa73-610d3d77dca2.jpg)
![批注 2022-03-15 102537](https://user-images.githubusercontent.com/58416716/158296665-f905a0c2-ba77-4ae5-b440-1648700dba23.jpg)

Reply:
hmm this looks incorrect to me. Even the other CS line.  What is the width if the PE3 off time?  ---_---

Reply:
![1647480348(1)](https://user-images.githubusercontent.com/58416716/158718195-6112974f-46a6-4e54-8cab-509df5c57860.png)
PE3(VDD_3V3_SENSORS_EN) is always on after initialization is complete.

Reply:
Our last version of the hardware is similar, when the chip is still ICM-20602.
![批注 2022-03-17 164347](https://user-images.githubusercontent.com/58416716/158771685-0cb937c9-07d9-465c-814f-205a40596132.jpg)
![批注 2022-03-17 164406](https://user-images.githubusercontent.com/58416716/158771705-efa5181b-8150-48b3-a5b3-7efd1b84752c.jpg)
![批注 2022-03-17 164447](https://user-images.githubusercontent.com/58416716/158771721-1e098d74-15a9-451d-896b-388ea8bdd882.jpg)

Reply:
@davids5 What do you think there is a problem?